### PR TITLE
feat: [CDS-80657]: remote selection of single and multi environments (Non Default Flow)

### PR DESCRIPTION
**Pull request by vikrant-harness**
### Summary

- Remote Linking of single environment with `gitBranch` being present in the YAML conditionally(pipeline studio + run pipeline form) 
- Remote Linking of multi environment with `gitBranch` being present in the YAML conditionally( pipeline studio + run pipeline form ) 
- Inline Edit for remote environment from the pipeline studio
- `Autoselect` environment logic when a new environment is added in pipeline studio

#### Screenshots

https://github.com/harness/harness-core-ui/assets/106532291/39812db2-c98c-4473-bed5-6c485682d43d

https://github.com/harness/harness-core-ui/assets/106532291/01227a4f-ba31-4e5d-a577-35838319fde7

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

File: src/modules/70-pipeline/components/FormMultiTypeEnvironmentField/FormMultiTypeEnvironmentField.tsx [hunk 1]
```diff
@@ -156,15 +173,35 @@ export function MultiTypeEnvironmentField(props: EnvironmentReferenceFieldProps)
...
-    }))
-    formik.setFieldValue(name, environments)
-    onMultiSelectChange(environments)
+    const gitMetadata: Record<string, string | undefined> = {}
+    const environments = envs.map((env: any) => {
+      const envId = getConnectorIdentifierWithScope(env.scope, env.identifier || '')
+      const branch = userSelectedBranches[envId]
+      const selectedEnvironmentBranch =
+        branch && branch !== defaultGitContextBranchPlaceholder
+          ? checkAndGetGitBranchForChildEntity(parentGitData, {
```
Comment (sunnykesh-harness) [inline src/modules/70-pipeline/components/FormMultiTypeEnvironmentField/FormMultiTypeEnvironmentField.tsx:187]: As per discussion, we can avoid this check and keep branch always if it is not a placeholder branch.

Comment (vikrant-harness) [reply src/modules/70-pipeline/components/FormMultiTypeEnvironmentField/FormMultiTypeEnvironmentField.tsx]: done

**Review by PratyushGar — APPROVED** [review]
LGTM

**Comment by vikrant-harness** [comment]
> LGTM. There are some instances of any, are they due to ongoing development and BE integration pending?

Will remove them as we progress towards the functionality completion

**Review by deepesh-ui — APPROVED** [review]

**Review by sunnykesh-harness — APPROVED** [review]

**Review by sonaligoyal-harness — APPROVED** [review]

**Comment by vikrant-harness** [comment]
Automation Ticket :- https://harness.atlassian.net/browse/CDS-83045

**Comment by sunnykesh-harness** [comment]
Need to fix Remote details alignment while selection as well.

**Pull request merged**